components/butterfly/src/server/mod.rs

```diff
@@ -513,7 +492,24 @@ impl Server {
                 Health::Departed
             );
 
+            // We need to mark this as "hot" in order to propagate it.
+            //
+            // TODO (CM): This exact code is present numerous places;
+            // factor it out to facilitate further code consolidation.
+            self.rumor_heat.start_hot_rumor(RumorKey::new(
+                RumorType::Member,
+                member.id.clone(),
+                "",
+            ));
+
             let check_list = self.member_list.check_list(&member.id);
+
+            // TODO (CM): Even though we marked the rumor as hot
+            // above, when we gossip, we send out the 5 "coolest but
+            // still warm" rumors. Sending to 10 members increases the
+            // chances that we'll get to this hot one now, but I don't
+            // think that we can strictly guarantee that this
+            // departure health update actually gets out in all cases.
```

Comment:
Perhaps change this magic number to a (ideally tunable) constant?

Reply:
I'll extract it to a constant, but leave it fixed for now. I'll also make a follow-up issue to make this tunable, along with all the other currently-hard-coded parameters that affect our gossip subsystem.